### PR TITLE
Fix gcc13 errors in llvm and cling 

### PR DESCRIPTION
**Pull request by linev**

### CODE REVIEW

**Comment by hahnjo** [comment]
Do we have to patch LLVM, ie is that an error or a warning? We're preparing changes to how we handle patches with LLVM, and in general we try to avoid them...

**Comment by linev** [comment]
> Do we have to patch LLVM

I getting error when compiling with gcc13:
```
interpreter/llvm/src/include/llvm/Support/Signals.h:119:24: error: ‘uintptr_t’ was not declared in this scope
  119 |   void CleanupOnSignal(uintptr_t Context);
```

**Comment by hahnjo** [comment]
> I getting error when compiling with gcc13:

Okay, because the PR title says "warning". This is not a warning...

**Comment by hahnjo** [comment]
FWIW the upstream commit is https://github.com/llvm/llvm-project/commit/ff1681ddb303223973653f7f5f3f3435b48a1983 - can we backport that instead of rolling our own change?

**Comment by linev** [comment]
> can we backport that

Of course. I can remove my commit from PR.

File: interpreter/llvm/src/include/llvm/Support/Signals.h [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #define LLVM_SUPPORT_SIGNALS_H
 
 #include <string>
```
Comment (vgvassilev) [inline interpreter/llvm/src/include/llvm/Support/Signals.h:17]: Can we probably backport https://github.com/llvm/llvm-project/commit/ff1681ddb303223973653f7f5f3f3435b48a1983 which is essentially the same but alphabetically sorted and clear that will not need to be put in the list of our patches to maintain.

**Review by vgvassilev — APPROVED** [review]
LGTM, see comment.

File: interpreter/llvm/src/include/llvm/Support/Signals.h (hunk 1 above)
Comment (hahnjo) [reply interpreter/llvm/src/include/llvm/Support/Signals.h:17]: I think we should still track it in our set of patches, but we will come back to this in today's team meeting

**Comment by linev** [comment]
I modify my commit that it compatible with changes in LLVM repository.

**Pull request merged**